### PR TITLE
[#IP-84] Change strict configuration to true 

### PR DESCRIPTION
**Pull request by fabriziopapi**
Set the stirct configuration to true.
Fixed the compilation errors caused by the new settings.

### CODE REVIEW

File: GetServiceByRevision/handler.ts [hunk 1]
```diff
@@ -136,9 +139,14 @@ export function GetServiceByRevision(
...
-  const middlewaresWrap = withRequestMiddlewares(
+  const middlewaresWrap = withRequestMiddlewares<
+    "IResponseErrorValidation",
+    "IResponseErrorValidation",
+    string & INonEmptyStringTag,
+    number & INonNegativeIntegerTag
+  >(
     RequiredParamMiddleware("serviceid", NonEmptyString),
-    RequiredParamMiddleware("version", IntegerFromString)
+    RequiredParamMiddleware("version", NonNegativeIntegerFromString)
```
Comment (balanza) [inline GetServiceByRevision/handler.ts:149]: Why are we introducing such change in this PR?

File: utils/appinsights.ts [hunk 2]
```diff
@@ -20,4 +21,4 @@ export const initTelemetryClient = (env = process.env) =>
             ).getOrElse(DEFAULT_SAMPLING_PERCENTAGE)
           })
         )
-        .getOrElse(undefined);
+        .fold(_ => undefined, identity);
```
Comment (balanza) [inline utils/appinsights.ts:24]: They are the same

**Review by balanza — CHANGES_REQUESTED** [review]
I see some implementation change that I would not expect from this PR:
* `NonNegativeInteger` introduced
* `"@pagopa/ts-commons`  installed

They seem reasonable, but did we need them for this task?

File: GetServiceByRevision/handler.ts (hunk 1 above)
Comment (fabriziopapi) [reply GetServiceByRevision/handler.ts:149]: The old code used the IntegerFromString io-style type to decode the input version (a string) into a number. The result is gived as input to a function parameter declared as non negative (using the INonNegativeIntegerTag type). In the strict mode, the compiler notice that IntegerFromString is not tagged properly, so I use the NonNegativeIntegerFromString type in order to properly decode the input string a non negative integer.

File: utils/appinsights.ts (hunk 2 above)
Comment (fabriziopapi) [reply utils/appinsights.ts]: Changed with proper error throws.

**Review by balanza — APPROVED** [review]
@fabriziopapi in last two commits I refactored a bit the introduced changes. Please review them and comment if anything is not sound to you.

A part from that, I'm happy with this PR, great job! Thanks a lot.

If you're fine, we can merge

**Comment by fabriziopapi** [comment]
I totally missed the IConfig type existence... good to know! :-)

Thank you, let's merge it!

**Pull request merged**